test/integration/gardenlet/vpaevictionrequirements/vpaevictionrequirements_test.go

```diff
@@ -0,0 +1,217 @@
+		targetDeployment               *appsv1.Deployment
+		maintenanceWindowNow           *gardencorev1beta1.MaintenanceTimeWindow
+		maintenanceWindowAlreadyPassed *gardencorev1beta1.MaintenanceTimeWindow
+		upscaleOnlyRequirement         = []*vpaautoscalingv1.EvictionRequirement{{
+			Resources:         []corev1.ResourceName{corev1.ResourceMemory, corev1.ResourceCPU},
+			ChangeRequirement: vpaautoscalingv1.TargetHigherThanRequests,
+		}}
+	)
+
+	BeforeEach(func() {
+		fakeClock.SetTime(time.Now())
+
+		maintenanceWindowNow = &gardencorev1beta1.MaintenanceTimeWindow{
+			Begin: time.Now().Format("150405-0700"),
+			End:   time.Now().Add(1 * time.Hour).Format("150405-0700"),
+		}
+		maintenanceWindowAlreadyPassed = &gardencorev1beta1.MaintenanceTimeWindow{
+			Begin: time.Now().Add(-3 * time.Hour).Format("150405-0700"),
+			End:   time.Now().Add(-2 * time.Hour).Format("150405-0700"),
+		}
+
+		targetDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-deployment",
+				Namespace: testNamespace.Name,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "test"}},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "test"}},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{Name: "test-container", Image: "my-nonexisting-image"},
+						},
+					},
+				},
+			},
+		}
+
+		vpa = &vpaautoscalingv1.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-vpa",
+				Namespace: testNamespace.Name,
+				Labels: map[string]string{
+					testID: testRunID,
+					v1beta1constants.LabelVPAEvictionRequirementsController: v1beta1constants.EvictionRequirementManagedByController,
+				},
+			},
+			Spec: vpaautoscalingv1.VerticalPodAutoscalerSpec{
+				TargetRef:    &autoscalingv1.CrossVersionObjectReference{Name: targetDeployment.Name, APIVersion: "apps/v1", Kind: "Deployment"},
+				UpdatePolicy: &vpaautoscalingv1.PodUpdatePolicy{EvictionRequirements: []*vpaautoscalingv1.EvictionRequirement{}},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Create deployment")
+		Expect(testClient.Create(ctx, targetDeployment)).To(Succeed())
+
+		DeferCleanup(func() {
+			By("Delete deployment")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, targetDeployment))).To(Succeed())
+		})
+
+		By("create VPA")
+		Expect(testClient.Create(ctx, vpa)).To(Succeed())
+
+		DeferCleanup(func() {
+			By("Delete VPA")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, vpa))).To(Succeed())
+		})
+
+	})
+
+	Context("VPA is annotated with downscale-in-maintenance-window-only", func() {
+		BeforeEach(func() {
+			metav1.SetMetaDataLabel(&vpa.ObjectMeta, v1beta1constants.LabelVPAEvictionRequirementDownscaleRestriction, v1beta1constants.EvictionRequirementInMaintenanceWindowOnly)
+		})
+		When("the Shoot is outside its maintenance window", func() {
```

Comment:
newline for readability

```suggestion

		When("the Shoot is outside its maintenance window", func() {
```

Reply:
done